### DISCUSSION TITLE
How would I `useSubmit()` without triggering a re-render before the action completes?

### DISCUSSION THREAD

**Discussion by svenvandescheur** [remix-run/react-router#11962] [Q&A]
I'm using the `useSubmit` hook to dispatch actions to perform state changes on an API (`const submit = useSubmit()`). I've noticed that my UI re-renders solely calling the returned `submit` function which interferers with my application (which it shouldn’t, I know).

I'm expecting my application NOT to re-render until the next completion of the loaders. Is this possible?

**Comment by mateusvahl** [comment]
@svenvandescheur did you solved this? I believe you can solve this with [shouldRevalidate](https://reactrouter.com/en/main/route/should-revalidate).

Just checking in to see you found other ways.

**Reply by svenvandescheur** [reply]
> @svenvandescheur did you solved this? I believe you can solve this with [shouldRevalidate](https://reactrouter.com/en/main/route/should-revalidate).
> 
> 
> 
> Just checking in to see you found other ways.

@mateusvahl thx for the link! Yes it would have probably solved my issue at the time, I did move a lot of the logic to custom hooks though to optimize the application.
